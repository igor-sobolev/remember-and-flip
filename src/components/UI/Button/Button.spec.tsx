import React from 'react';

import { Button } from '.';
import { shallow, ShallowWrapper } from 'enzyme';

const givenProps = {
  onClick: jest.fn()
};

describe('Given Button component', () => {
  describe('when it is rendered', () => {
    let wrapper: ShallowWrapper;
    beforeEach(() => {
      wrapper = shallow(<Button {...givenProps}>Test button</Button>);
    });

    it('should match the snapshot', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    describe('when it is clicked', () => {
      beforeEach(() => {
        wrapper.simulate('click');
      });

      it('should call onClick', () => {
        expect(givenProps.onClick).toHaveBeenCalled();
      });
    });
  });
});
