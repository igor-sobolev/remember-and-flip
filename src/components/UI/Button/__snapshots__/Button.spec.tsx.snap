// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Button component when it is rendered should match the snapshot 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
>
  Test button
</button>
`;
